### ISSUE TITLE
Add shizuku support for ADB privilege

### ISSUE THREAD

**Issue by Cyberavater** [MuntashirAkon/AppManager#55]
It would be nice if non-root user could use Shizuku for ADB privilege
https://github.com/RikkaApps/Shizuku

**Comment by MuntashirAkon** [comment]
I'm not sure if I will do that. RikkaApps doesn't have a license for Shizuku and the related issue (RikkaApps/Shizuku/issues/56) is closed without any comment. They've also declined to publish their app on F-Droid (RikkaApps/Shizuku/issues/69) which they can't because the app doesn't have any license (Even if I want, I can't even fork the app to provide a release on F-Droid for the same reason). They do have a license for their API but, I have to say, the whole thing is a bit shady and being a strong supported of freedom, I'm not going to support Shizuku.

I do have plan to work more on ADB and reflection to make the process a bit easier. In future, I'll also support installing App Manager as a system app (which can be built with custom ROMs), which won't be requiring any root for most tasks.

In short, due to Shizuku's policy and my roadmaps, I can't support Shizuku.

**Comment by Cyberavater** [comment]
Understandable.

**Comment by MuntashirAkon** [comment]
A more formal answer is given here: https://muntashirakon.github.io/AppManager/faq/misc.html#any-plans-for-shizuku

**Comment by Pandapip1** [comment]
Shizuku is now licensed under Apache-2. I suggest this or #55 be implemented.

**Comment by MuntashirAkon** [comment]
> Shizuku is now licensed under Apache-2. I suggest this or #55 be reopened

Shizuku is still [non-free](https://apt.izzysoft.de/fdroid/index/apk/moe.shizuku.privileged.api) due to its ToS. OSI license is not enough to judge if a software is free. It must not adhere to any additional terms that might make it non-free.

**Comment by Pandapip1** [comment]
Fair enough.

**Comment by e-t-l** [comment]
> A more formal answer is given here: https://muntashirakon.github.io/AppManager/faq/misc.html#any-plans-for-shizuku

This leads to an empty 404 error page


> > Shizuku is now licensed under Apache-2. I suggest this or #55 be reopened
> 
> Shizuku is still [non-free](https://apt.izzysoft.de/fdroid/index/apk/moe.shizuku.privileged.api) due to its ToS. OSI license is not enough to judge if a software is free. It must not adhere to any additional terms that might make it non-free.

I won't pretend to understand the license legalese, and I don't really know what "non-free" means in this context, but the [Shizuku-API](https://github.com/RikkaApps/Shizuku-API/blob/master/LICENSE), which is what would actually need to be included in the app, has a pretty straightforward and permissive MIT license as far as I can tell. Is that not sufficient? The best other non-root option for AppManager, wireless debugging mode, is pretty inconvenient since it requires a constant wifi connection. Shizuku avoids this issue by persisting beyond wifi disconnect.

**Comment by MuntashirAkon** [comment]
Latest explanation is available here: https://github.com/amcommunity/TipsAndTricks/issues/14

**Comment by mariush444** [comment]
anyway, if not shizuke then it would be nicw to have block tracker and freez apk without root

**Comment by MuntashirAkon** [comment]
> anyway, if not shizuke then it would be nicw to have block tracker and freez apk without root

You cannot block trackers with ADB, regardless whether you use Shizuku or not. This is because Shell doesn't have enough permissions to block tracker components. You can already freeze/unfreeze apps with ADB.

**Comment by Thorfinn66** [comment]
It does say this on their github page:

License

The code for this project is available under the Apache-2.0 license.

**Comment by goddaneel** [comment]
shizuku has revised the license ambiguities and the code for the project is now available downstream as normal.
I'm not sure if you still need it at this time, but I felt the need to provide you with this information.

> All code files in this project are licensed under Apache 2.0
> Under Apache 2.0 section 6, specifically:
>    You are FORBIDDEN to use manager/src/main/res/mipmap*/ic_launcher*.png image files, unless for displaying Shizuku itself.
>    You are FORBIDDEN to distribute the apk that use Shizuku as name or use moe.shizuku.privileged.api as application id or declare moe.shizuku.manager.permission.* permission to any store (IBNLT Google Play Store, F-Droid, Amazon Appstore etc.).

https://github.com/RikkaApps/Shizuku/commit/7bd50a24189446114d2fd669842bbdc720411142

**Comment by Coldblackice** [comment]
> > shizuku has revised the license ambiguities and the code for the project is now available downstream as normal. I'm not sure if you still need it at this time, but I felt the need to provide you with this information.
> > > All code files in this project are licensed under Apache 2.0
> > 
> > 
> > > Under Apache 2.0 section 6, specifically:
> > 
> > 
> > > * You are **FORBIDDEN** to use `manager/src/main/res/mipmap*/ic_launcher*.png` image files, unless for displaying Shizuku itself.
> > 
> > 
> > > * You are **FORBIDDEN** to distribute the apk that use Shizuku as name
> > >   or use `moe.shizuku.privileged.api` as application id or declare `moe.shizuku.manager.permission.*` permission
> > >   to any store (IBNLT Google Play Store, F-Droid, Amazon Appstore etc.).
> > 
> > 
> > [RikkaApps/Shizuku@7bd50a2](https://github.com/RikkaApps/Shizuku/commit/7bd50a24189446114d2fd669842bbdc720411142)
> 
> Thanks for bringing this to my attention. This indeed looks FOSS-compliant to me. I'll update the docs accordingly.

Awesome news! Does this mean you're open to considering implementing Shizuku access for ADB in AppManager?

**Comment by om3ga6400** [comment]
if this was added it would make app manager so much less painful to start up every time

**Comment by TechMan205** [comment]
yea shizuku is now truly a very big deal now adays its no joke at all so if shizuku support is added for anything the app will need it will just gain much more traction than it is getting by it self and get many many users cuz it will be listed as a shizuku supported app in the list of most shizuku supported apps

**Comment by marek22k** [comment]
Would [Dhizuku](https://github.com/iamr0s/Dhizuku) be an alternative?

**Comment by Tharow** [comment]
> Would [Dhizuku](https://github.com/iamr0s/Dhizuku) be an alternative?

@marek22k Having Tried it before, it sorta works the only problem is that AM requires a lot of direct access it works better with Test DPC and using delegation of permissions. the problem is that AM works best when it itself is the device owner, allows greater control of the install process.
If you want i could help you get started on implementing that, but I personally have my own things I'm working on.

Personally its great to use AM like this, its what i use for my daily driver as it's persistent across restarts

**Comment by Skorpion96** [comment]
Some time ago I got shizuku working on app manager terminal by doing some modifications to rish btw

![Image](https://github.com/user-attachments/assets/2a804336-782b-41c4-a4a2-ad5b62730648)

**Comment by zer0boi** [comment]
It's been a while so I'm here to ask on everyone else's behalf, will shizuku implantation be introduced soon? 👀

I'm sure many share my frustration of not wanting to require WiFi connection to use the app, so i hope this is considered

--

Edit: the docs say

> 4.3.4 [Any plans for Shizuku?](https://muntashir.dev/AppManager/en/#toc:subsec:faq:shizuku-support)
App Manager’s use of hidden API and privileged code execution is now much more complex and cannot be integrated with other third party apps such as [Shizuku](https://shizuku.rikka.app/). Here are some reasons for not considering Shizuku (which now has Apache 2.0 license) for App Manager:

> 1 Shizuku was initially non-free which led me to use a similar approach for App Manager to support both root and ADB

> 2 App Manager already supports both ADB and root which in some cases is more capable than Shizuku

> 3 Relying on a third-party app for the major functionalities is not a good design choice

> 4 Integration of Shizuku will increase the complexity of App Manager.

However, non rooted devices will still really need a way beside wifi dependent ADB so can't it be really implemented as an option?

**Comment by Tharow** [comment]
@marek22k I have a version of AM that supports DPC if you want to try it https://github.com/MuntashirAkon/AppManager/pull/1858 it Requires that AM be the Device Owner tho.

@zer0boi That Depends entirely on MuntashirAkon. I could have  pull request for it by end of day or tomorrow at the latest 
The biggest issue is that AM would only use Shizuku to start the shell server anyway, which you could do yourself with the included commands in the OpsModePref setting page making the feature redundant at this point.

Also AM has the same startup options as Shizuku? What are you doing with Shizuku that's different then AM?

**Comment by zer0boi** [comment]
@Tharow wait, you're saying that just like shizuku you can start AM via adb once and as long as you don't restart your device then it will still work on elevated privileges anywhere?

I thought it required adb every single time, i believe that's the reason i and some others here were requesting it. Personally i remember it not working after turning off adb once so maybe that was a separate bug?

Anyways if that's the case then mb, shizuku won't be that necessary anymore...

*well* it would still be nice since a basic shizuku feature to simply start the service instead of adb would mean that we won't need to start both shizuku and AM each time we reboot our phones, we would simply start only shizuku at reboot and it will auto cover all special apps including AM. That'll be a nice little quality of life feature

**Comment by Tharow** [comment]
> @Tharow wait, you're saying that just like shizuku you can start AM via adb once and as long as you don't restart your device then it will still work on elevated privileges anywhere?
> 
> I thought it required adb every single time, i believe that's the reason i and some others here were requesting it. Personally i remember it not working after turning off adb once so maybe that was a separate bug?
> 
> Anyways if that's the case then mb, shizuku won't be that necessary anymore...
> 
> *well* it would still be nice since a basic shizuku feature to simply start the service instead of adb would mean that we won't need to start both shizuku and AM each time we reboot our phones, we would simply start only shizuku at reboot and it will auto cover all special apps including AM. That'll be a nice little quality of life feature

Well you can start AM from a shizuku powered terminal. Using the mode of ops commands. 
And yes the server keeps running though it reconnection is sometimes finicky I would advise enabling the background process for AM (under Privacy in settings  for reasons no one understands) 
To prevent AM from losing connection to its internal server. 

Realised I should explain, AM creates a internal server, under the shell user and is persistent even with lack of internet. As long as the server is running AM has adb powers.
That internal server doesn't communicate over internet.

As a warning some ROMs kill the all processes associated with adb once all network sources are gone (not including cell, but including USB which is technically a network connection.)

**Comment by zer0boi** [comment]
> Well you can start AM from a shizuku powered terminal. Using the mode of ops commands. And yes the server keeps running though it reconnection is sometimes finicky I would advise enabling the background process for AM (under Privacy in settings for reasons no one understands) To prevent AM from losing connection to its internal server.
> 
> Realised I should explain, AM creates a internal server, under the shell user and is persistent even with lack of internet. As long as the server is running AM has adb powers. That internal server doesn't communicate over internet.
> 
> As a warning some ROMs kill the all processes associated with adb once all network sources are gone (not including cell, but including USB which is technically a network connection.)

Ah I'm trying to avoid having background activity generally. Plus my phone brand (Redmi) has an annoyingly unchangeable (as far as i tried) procrasses killer so maybe it's this or that which prevented AM from working properly for me

But Anyways, you said something about being able to start AM via a shizuku terminal? And i mean start it with elevated privileges or kicks off it's server or something without needing adb and wifi? If there's such a command then I'd appreciate having it. Tho to make this somewhat automatic by adding a shixuku feature in AM itself would still be neet

**Comment by Tharow** [comment]
> 
> Ah I'm trying to avoid having background activity generally. Plus my phone brand (Redmi) has an annoyingly unchangeable (as far as i tried) procrasses killer so maybe it's this or that which prevented AM from working properly for me
> 
> But Anyways, you said something about being able to start AM via a shizuku terminal? And i mean start it with elevated privileges or kicks off it's server or something without needing adb and wifi? If there's such a command then I'd appreciate having it. Tho to make this somewhat automatic by adding a shixuku feature in AM itself would still be neet


In mode of Ops page commands are listed try both as one might not work for you.
![Screenshot_20260306-065619.png](https://github.com/user-attachments/assets/be553109-1fd6-4e07-9808-ae58d30c1167)

Please ignore the DPC mode command in the picture.

**Comment by e-t-l** [comment]
> some ROMs kill the all processes associated with adb once all network sources are gone 

Where can I research into which ROMs do this? I think this might be related to a different issue

**Comment by zer0boi** [comment]
@Tharow thanks, i don't know how i missed that command section before but it works 👍
I don't know if the other users would still need the auto shizuku step but this does it for me

**Comment by MuntashirAkon** [comment]
From my discussion with people experienced in interpreting licensing, it appears Shizuku app is still not considered to be FOSS compliant. Like I have announced in a previous update, unless there is a general purpose solution (like Open Keychain's) where any fork can use the same API and the users are allowed to choose which fork (or the original) they like to use (provided there are independent forks), we cannot consider Shizuku to be an option. Instead, my recommendation for ADB users is that if you face any issues, open a new issue or provide a productive feedback to the existing ones. Most problems in the world are actually solvable if we come close together. Many other projects also closely follow App Manager's handling of ADB (for being the only fully open source solution). So, you not only help App Manager but those other projects too.

App Manager is built in the spirit of the open source software, and it extensively made use of open source code written by other developers during its initial phase. I know many users do not care about things like this, but if there had been no true open source software, you would have never had Android in first place (and certainly no custom ROMs would've been possible because nobody would release the kernel source code).

**Comment by Skorpion96** [comment]
> From my discussion with people experienced in interpreting licensing, it appears Shizuku app is still not considered to be FOSS compliant. Like I have announced in a previous update, unless there is a general purpose solution (like Open Keychain's) where any fork can use the same API and the users are allowed to choose which fork (or the original) they like to use (provided there are independent forks), we cannot consider Shizuku to be an option. Instead, my recommendation for ADB users is that if you face any issues, open a new issue or provide a productive feedback to the existing ones. Most problems in the world are actually solvable if we come close together. Many other projects also closely follow App Manager's handling of ADB (for being the only fully open source solution). So, you not only help App Manager but those other projects too.
> 
> App Manager is built in the spirit of the open source software, and it extensively made use of open source code written by other developers during its initial phase. I know many users do not care about things like this, but if there had been no true open source software, you would have never had Android in first place (and certainly no custom ROMs would've been possible because nobody would release the kernel source code).

I mean, ok but solutions exist, example app manager could curl rish and sed it to add it's package name, then run it to acquire the privs. Since it's being crawled from a server the app doesn't have inside code about shizuku.

**Comment by MuntashirAkon** [comment]
> app manager could curl rish and sed it to add it's package name, then run it to acquire the privs. Since it's being crawled from a server the app doesn't have inside code about shizuku.

When you `curl` into rish, you automatically introduce a non-free dependency, because rish itself is non-free.

**Comment by Skorpion96** [comment]
> > app manager could curl rish and sed it to add it's package name, then run it to acquire the privs. Since it's being crawled from a server the app doesn't have inside code about shizuku.
> 
> When you `curl` into rish, you automatically introduce a non-free dependency, because rish itself is non-free.

Then shizuku itself is a paradox, to use it you must use rish but that might introduce non free code, i wonder if could be a license violation.

Anyway, what about some system shell support? Like we run a reverse system shell and app manager connects and would be able to use system privileges to do actions as system (hopefully not limited to a shell).
